New download paths for github-based depends don't seem to work when the package version is a git hash

For `native_cctools`, we have
```
package=native_cctools
$(package)_version=55562e4073dea0fbfd0b20e0bf69ffe6390c7f97
$(package)_download_path=https://github.com/tpoechtrager/cctools-port/archive/refs/tags
$(package)_file_name=$($(package)_version).tar.gz
$(package)_sha256_hash=e51995a843533a3dac155dd0c71362dd471597a2d23f13dff194c6285362f875
```
in `packages/native_cctools.mk`, but if we try and fetch what I believe to be the URL, it fails with a 404:
```
sasha_work@seasonal-dream compares % wget https://github.com/tpoechtrager/cctools-port/archive/refs/tags/55562e4073dea0fbfd0b20e0bf69ffe6390c7f97.tar.gz
--2023-02-08 13:48:29--  https://github.com/tpoechtrager/cctools-port/archive/refs/tags/55562e4073dea0fbfd0b20e0bf69ffe6390c7f97.tar.gz
Resolving github.com (github.com)... 192.30.255.113
Connecting to github.com (github.com)|192.30.255.113|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://codeload.github.com/tpoechtrager/cctools-port/tar.gz/refs/tags/55562e4073dea0fbfd0b20e0bf69ffe6390c7f97 [following]
--2023-02-08 13:48:29--  https://codeload.github.com/tpoechtrager/cctools-port/tar.gz/refs/tags/55562e4073dea0fbfd0b20e0bf69ffe6390c7f97
Resolving codeload.github.com (codeload.github.com)... 192.30.255.121
Connecting to codeload.github.com (codeload.github.com)|192.30.255.121|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2023-02-08 13:48:29 ERROR 404: Not Found.

sasha_work@seasonal-dream compares % 
```

Likewise with `tl_expected`:

```
package=tl_expected
$(package)_version=1.0.1
$(package)_download_path=https://github.com/TartanLlama/expected/archive/refs/tags
$(package)_download_file=96d547c03d2feab8db64c53c3744a9b4a7c8f2c5.tar.gz
$(package)_file_name=tl-expected-1.0.1.tar.gz
$(package)_sha256_hash=64901df1de9a5a3737b331d3e1de146fa6ffb997017368b322c08f45c51b90a7
```

gets us:
```
sasha_work@seasonal-dream compares % wget https://github.com/TartanLlama/expected/archive/refs/tags/96d547c03d2feab8db64c53c3744a9b4a7c8f2c5.tar.gz
--2023-02-08 13:49:13--  https://github.com/TartanLlama/expected/archive/refs/tags/96d547c03d2feab8db64c53c3744a9b4a7c8f2c5.tar.gz
Resolving github.com (github.com)... 192.30.255.113
Connecting to github.com (github.com)|192.30.255.113|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://codeload.github.com/TartanLlama/expected/tar.gz/refs/tags/96d547c03d2feab8db64c53c3744a9b4a7c8f2c5 [following]
--2023-02-08 13:49:13--  https://codeload.github.com/TartanLlama/expected/tar.gz/refs/tags/96d547c03d2feab8db64c53c3744a9b4a7c8f2c5
Resolving codeload.github.com (codeload.github.com)... 192.30.255.121
Connecting to codeload.github.com (codeload.github.com)|192.30.255.121|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2023-02-08 13:49:14 ERROR 404: Not Found.

sasha_work@seasonal-dream compares % 
```

The old path (with the `refs/tags` excised) works for both:

```
sasha_work@seasonal-dream compares % wget https://github.com/TartanLlama/expected/archive/96d547c03d2feab8db64c53c3744a9b4a7c8f2c5.tar.gz 
--2023-02-08 13:49:51--  https://github.com/TartanLlama/expected/archive/96d547c03d2feab8db64c53c3744a9b4a7c8f2c5.tar.gz
Resolving github.com (github.com)... 192.30.255.113
Connecting to github.com (github.com)|192.30.255.113|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://codeload.github.com/TartanLlama/expected/tar.gz/96d547c03d2feab8db64c53c3744a9b4a7c8f2c5 [following]
--2023-02-08 13:49:51--  https://codeload.github.com/TartanLlama/expected/tar.gz/96d547c03d2feab8db64c53c3744a9b4a7c8f2c5
Resolving codeload.github.com (codeload.github.com)... 192.30.255.121
Connecting to codeload.github.com (codeload.github.com)|192.30.255.121|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [application/x-gzip]
Saving to: ‘96d547c03d2feab8db64c53c3744a9b4a7c8f2c5.tar.gz.2’

96d547c03d2feab8db64c53c3744a9b4a7c8f2     [ <=>                                                                        ]  21.48K  --.-KB/s    in 0.001s  

2023-02-08 13:49:51 (33.1 MB/s) - ‘96d547c03d2feab8db64c53c3744a9b4a7c8f2c5.tar.gz.2’ saved [21994]

sasha_work@seasonal-dream compares % wget https://github.com/tpoechtrager/cctools-port/archive/55562e4073dea0fbfd0b20e0bf69ffe6390c7f97.tar.gz 
--2023-02-08 13:50:01--  https://github.com/tpoechtrager/cctools-port/archive/55562e4073dea0fbfd0b20e0bf69ffe6390c7f97.tar.gz
Resolving github.com (github.com)... 192.30.255.113
Connecting to github.com (github.com)|192.30.255.113|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://codeload.github.com/tpoechtrager/cctools-port/tar.gz/55562e4073dea0fbfd0b20e0bf69ffe6390c7f97 [following]
--2023-02-08 13:50:01--  https://codeload.github.com/tpoechtrager/cctools-port/tar.gz/55562e4073dea0fbfd0b20e0bf69ffe6390c7f97
Resolving codeload.github.com (codeload.github.com)... 192.30.255.121
Connecting to codeload.github.com (codeload.github.com)|192.30.255.121|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [application/x-gzip]
Saving to: ‘55562e4073dea0fbfd0b20e0bf69ffe6390c7f97.tar.gz.2’

55562e4073dea0fbfd0b20e0bf69ffe6390c7f     [   <=>                                                                      ]   2.72M  5.82MB/s    in 0.5s    

2023-02-08 13:50:02 (5.82 MB/s) - ‘55562e4073dea0fbfd0b20e0bf69ffe6390c7f97.tar.gz.2’ saved [2851747]

sasha_work@seasonal-dream compares % 
```

I believe this change was introduced with PR #6384; and there might be something I am missing (maybe the URL derivation process is not what I think it is).

Adding `native_cctools` to the `native_packages` line in `packages/packages.mk` to force it to install anyway on a linux machine fails with 4 404s:

```
sasha_work@myouren:~/src/zcash/depends$ make 
Checksum missing or mismatched for native_cctools source. Forcing re-download.
Fetching native_cctools...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0    14    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 2784k  100 2784k    0     0  21.4M      0 --:--:-- --:--:-- --:--:-- 21.5M
/home/sasha_work/src/zcash/depends/work/download/native_cctools-55562e4073dea0fbfd0b20e0bf69ffe6390c7f97/55562e4073dea0fbfd0b20e0bf69ffe6390c7f97.tar.gz.temp: OK
Fetching native_cctools...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0    14    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404
make: *** [funcs.mk:278: /home/sasha_work/src/zcash/depends/sources/download-stamps/.stamp_fetched-native_cctools-55562e4073dea0fbfd0b20e0bf69ffe6390c7f97.tar.gz.hash] Error 22
```

* github 404 for `https://github.com/tpoechtrager/cctools-port/archive/refs/tags/55562e4073dea0fbfd0b20e0bf69ffe6390c7f97.tar.gz` (native_cctools itself)
* downloads.z.cash 404 for `https://download.z.cash/depends-sources/55562e4073dea0fbfd0b20e0bf69ffe6390c7f97.tar.gz` (likewise native_cctools)

and also two 404s for `libtapi` (though that's not clear from the output, you have to run `make V=1` to see that they're different URLs and look at `packages/native_cctools.mk` to see what package name they're for)

* github 404 for `https://github.com/tpoechtrager/apple-libtapi/archive/refs/tags/3efb201881e7a76a21e0554906cf306432539cef.tar.gz`
* and downloads.z.cash 404 for `https://download.z.cash/depends-sources/3efb201881e7a76a21e0554906cf306432539cef.tar.gz`

--- a/depends/packages/native_cctools.mk
+++ b/depends/packages/native_cctools.mk
@@ -1,7 +1,8 @@
 package=native_cctools
 $(package)_version=55562e4073dea0fbfd0b20e0bf69ffe6390c7f97
 $(package)_download_path=https://github.com/tpoechtrager/cctools-port/archive
-$(package)_file_name=$($(package)_version).tar.gz
+$(package)_download_file=$($(package)_version).tar.gz
+$(package)_file_name=$(package)_$($(package)_version).tar.gz
 $(package)_sha256_hash=e51995a843533a3dac155dd0c71362dd471597a2d23f13dff194c6285362f875
 $(package)_build_subdir=cctools
 $(package)_dependencies=native_clang
@@ -10,7 +11,7 @@ $(package)_patches=ignore-otool.diff
 $(package)_libtapi_version=3efb201881e7a76a21e0554906cf306432539cef
 $(package)_libtapi_download_path=https://github.com/tpoechtrager/apple-libtapi/archive
 $(package)_libtapi_download_file=$($(package)_libtapi_version).tar.gz
-$(package)_libtapi_file_name=$($(package)_libtapi_version).tar.gz
+$(package)_libtapi_file_name=$(package)_libtapi_$($(package)_libtapi_version).tar.gz
 $(package)_libtapi_sha256_hash=380c1ca37cfa04a8699d0887a8d3ee1ad27f3d08baba78887c73b09485c0fbd3
 
 $(package)_extra_sources += $($(package)_libtapi_file_name)

--- a/depends/packages/native_cctools.mk
+++ b/depends/packages/native_cctools.mk
@@ -1,6 +1,6 @@
 package=native_cctools
 $(package)_version=55562e4073dea0fbfd0b20e0bf69ffe6390c7f97
-$(package)_download_path=https://github.com/tpoechtrager/cctools-port/archive/refs/tags
+$(package)_download_path=https://github.com/tpoechtrager/cctools-port/archive
 $(package)_file_name=$($(package)_version).tar.gz
 $(package)_sha256_hash=e51995a843533a3dac155dd0c71362dd471597a2d23f13dff194c6285362f875
 $(package)_build_subdir=cctools
@@ -8,7 +8,7 @@ $(package)_dependencies=native_clang
 $(package)_patches=ignore-otool.diff
 
 $(package)_libtapi_version=3efb201881e7a76a21e0554906cf306432539cef
-$(package)_libtapi_download_path=https://github.com/tpoechtrager/apple-libtapi/archive/refs/tags
+$(package)_libtapi_download_path=https://github.com/tpoechtrager/apple-libtapi/archive
 $(package)_libtapi_download_file=$($(package)_libtapi_version).tar.gz
 $(package)_libtapi_file_name=$($(package)_libtapi_version).tar.gz
 $(package)_libtapi_sha256_hash=380c1ca37cfa04a8699d0887a8d3ee1ad27f3d08baba78887c73b09485c0fbd3

--- a/depends/packages/tl_expected.mk
+++ b/depends/packages/tl_expected.mk
@@ -1,8 +1,8 @@
 package=tl_expected
-$(package)_version=1.0.1
+$(package)_version=96d547c03d2feab8db64c53c3744a9b4a7c8f2c5
 $(package)_download_path=https://github.com/TartanLlama/expected/archive
-$(package)_download_file=96d547c03d2feab8db64c53c3744a9b4a7c8f2c5.tar.gz
-$(package)_file_name=tl-expected-1.0.1.tar.gz
+$(package)_download_file=$($(package)_version).tar.gz
+$(package)_file_name=$(package)_$($(package)_version).tar.gz
 $(package)_sha256_hash=64901df1de9a5a3737b331d3e1de146fa6ffb997017368b322c08f45c51b90a7
 $(package)_patches=remove-undefined-behaviour.diff
 

--- a/depends/packages/tl_expected.mk
+++ b/depends/packages/tl_expected.mk
@@ -1,6 +1,6 @@
 package=tl_expected
 $(package)_version=1.0.1
-$(package)_download_path=https://github.com/TartanLlama/expected/archive/refs/tags
+$(package)_download_path=https://github.com/TartanLlama/expected/archive
 $(package)_download_file=96d547c03d2feab8db64c53c3744a9b4a7c8f2c5.tar.gz
 $(package)_file_name=tl-expected-1.0.1.tar.gz
 $(package)_sha256_hash=64901df1de9a5a3737b331d3e1de146fa6ffb997017368b322c08f45c51b90a7

